src/App.jsx

```diff
@@ -1,35 +1,98 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useContext, useState, useEffect } from "react";
+import "./App.css";
+import { useNavigate, Outlet } from "react-router-dom";
+import { ThemeContext } from "./ThemeContext";
+import store from "./store/store";
 
 function App() {
-  const [count, setCount] = useState(0)
-
-  return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+	const [favsLength, setFavsLength] = useState(0);
+
+	const navigate = useNavigate();
+
+	useEffect(() => {
```

Comment:
`useSelector` is a hook that will do this for you, and you could also eliminate your `useStateHook`

`const favsLength = useSelector((state) => state.pokemon.length)`